### PR TITLE
[branch/v8] Backport #8840

### PR DESCRIPTION
**Pull request by xacrimon**
Clean backport, no conflicts or changes.
Also includes my gRPC fixes for Apple Silicon.

### CODE REVIEW

**Review by rosstimothy — APPROVED** [review]

**Review by greedy52 — APPROVED** [review]

**Comment by russjones** [comment]
@xacrimon Can you test and merge this PR?

**Comment by xacrimon** [comment]
@russjones waiting on docs/super approval

**Review by ptgott — APPROVED** [review]

**Pull request merged**
